### PR TITLE
Configuration of new 8ch switch

### PR DESCRIPTION
**Pull request by bszczepanik**
I would like to add configuration for this new 8ch switch https://www.alibaba.com/product-detail/Tuya-smart-wifi-zigbee-Bluetooth-4CH_1600834099744.html

Look similar like TS0601_switch_8 but this one has different manufacturerName and last two Tuya data points.

### CODE REVIEW

**Comment by bszczepanik** [comment]
I found na issue with switch multiply switches. Seperate request messages is sending in same time. Device send readback as switches are in expected possitions. But they aren't. I think the workaround is to set delay between request messages. How can I do this?

**Comment by Koenkk** [comment]
Could you provide the debug log when sending multiple commands at once and it doesn't work? Currently we don't support this artificial delay.

See https://www.zigbee2mqtt.io/guide/usage/debug.html on how to enable debug logging.

**Comment by Koenkk** [comment]
Note to self; after this merge https://github.com/Koenkk/zigbee2mqtt.io/pull/2348

**Comment by bszczepanik** [comment]
That is debug log, when channel 7 stucked
`Debug 2023-11-24 12:20:13 Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l1/set' with data 'ON'
Debug 2023-11-24 12:20:13 Publishing 'set' 'state' to 'Underfloor heating switch'
Info 2023-11-24 12:20:13 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":61,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"OFF","state_l7":"OFF","state_l8":"OFF"}'
Debug 2023-11-24 12:20:13 Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l6/set' with data 'ON'
Debug 2023-11-24 12:20:13 Publishing 'set' 'state' to 'Underfloor heating switch'
Debug 2023-11-24 12:20:13 Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l7/set' with data 'ON'
Debug 2023-11-24 12:20:13 Publishing 'set' 'state' to 'Underfloor heating switch'
Info 2023-11-24 12:20:13 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":61,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"OFF","state_l8":"OFF"}'
Debug 2023-11-24 12:20:13 Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[1],"type":"Buffer"},"datatype":1,"dp":1}],"seq":18432}' from endpoint 1 with groupID 0
Info 2023-11-24 12:20:13 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":72,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"OFF","state_l8":"OFF"}'
Info 2023-11-24 12:20:13 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":72,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"ON","state_l8":"OFF"}'
Debug 2023-11-24 12:20:13 Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[1],"type":"Buffer"},"datatype":1,"dp":6}],"seq":18688}' from endpoint 1 with groupID 0
Info 2023-11-24 12:20:13 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":76,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"ON","state_l8":"OFF"}'
Debug 2023-11-24 12:20:13 Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[1],"type":"Buffer"},"datatype":1,"dp":7}],"seq":18944}' from endpoint 1 with groupID 0
Info 2023-11-24 12:20:13 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":76,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"ON","state_l8":"OFF"}'
Debug 2023-11-24 12:20:18 Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l1/set' with data 'OFF'
Debug 2023-11-24 12:20:18 Publishing 'set' 'state' to 'Underfloor heating switch'
Info 2023-11-24 12:20:18 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":76,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"ON","state_l8":"OFF"}'
Debug 2023-11-24 12:20:18 Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l6/set' with data 'OFF'
Debug 2023-11-24 12:20:18 Publishing 'set' 'state' to 'Underfloor heating switch'
Debug 2023-11-24 12:20:18 Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l7/set' with data 'OFF'
Debug 2023-11-24 12:20:18 Publishing 'set' 'state' to 'Underfloor heating switch'
Debug 2023-11-24 12:20:18 Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0],"type":"Buffer"},"datatype":1,"dp":1}],"seq":19200}' from endpoint 1 with groupID 0
Info 2023-11-24 12:20:18 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":65,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"ON","state_l8":"OFF"}'
Info 2023-11-24 12:20:18 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":65,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"OFF","state_l7":"ON","state_l8":"OFF"}'
Info 2023-11-24 12:20:18 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":65,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"OFF","state_l7":"OFF","state_l8":"OFF"}'
Debug 2023-11-24 12:20:18 Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0],"type":"Buffer"},"datatype":1,"dp":6}],"seq":19456}' from endpoint 1 with groupID 0
Info 2023-11-24 12:20:18 MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":65,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"OFF","state_l7":"OFF","state_l8":"OFF"}'`

And that when I put 100 ms delay in Home Assistant, that's way everything worked
`Debug 2023-11-24 12:22:25Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l1/set' with data 'ON'
Debug 2023-11-24 12:22:25Publishing 'set' 'state' to 'Underfloor heating switch'
Info 2023-11-24 12:22:25MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":76,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"OFF","state_l7":"OFF","state_l8":"OFF"}'
Debug 2023-11-24 12:22:26Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[1],"type":"Buffer"},"datatype":1,"dp":1}],"seq":19712}' from endpoint 1 with groupID 0
Info 2023-11-24 12:22:26MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":69,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"OFF","state_l7":"OFF","state_l8":"OFF"}'
Debug 2023-11-24 12:22:26Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l6/set' with data 'ON'
Debug 2023-11-24 12:22:26Publishing 'set' 'state' to 'Underfloor heating switch'
Info 2023-11-24 12:22:26MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":69,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"OFF","state_l8":"OFF"}'
Debug 2023-11-24 12:22:26Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l7/set' with data 'ON'
Debug 2023-11-24 12:22:26Publishing 'set' 'state' to 'Underfloor heating switch'
Debug 2023-11-24 12:22:26Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[1],"type":"Buffer"},"datatype":1,"dp":6}],"seq":19968}' from endpoint 1 with groupID 0
Info 2023-11-24 12:22:26MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":72,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"OFF","state_l8":"OFF"}'
Info 2023-11-24 12:22:26MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":72,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"ON","state_l8":"OFF"}'
Debug 2023-11-24 12:22:26Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[1],"type":"Buffer"},"datatype":1,"dp":7}],"seq":20224}' from endpoint 1 with groupID 0
Info 2023-11-24 12:22:26MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":69,"state_l1":"ON","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"ON","state_l8":"OFF"}'
Debug 2023-11-24 12:22:31Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l1/set' with data 'OFF'
Debug 2023-11-24 12:22:31Publishing 'set' 'state' to 'Underfloor heating switch'
Info 2023-11-24 12:22:31MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":69,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"ON","state_l8":"OFF"}'
Debug 2023-11-24 12:22:31Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0],"type":"Buffer"},"datatype":1,"dp":1}],"seq":20480}' from endpoint 1 with groupID 0
Info 2023-11-24 12:22:31MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":72,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"ON","state_l7":"ON","state_l8":"OFF"}'
Debug 2023-11-24 12:22:31Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l6/set' with data 'OFF'
Debug 2023-11-24 12:22:31Publishing 'set' 'state' to 'Underfloor heating switch'
Info 2023-11-24 12:22:31MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":72,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"OFF","state_l7":"ON","state_l8":"OFF"}'
Debug 2023-11-24 12:22:31Received MQTT message on 'zigbee2mqtt/Underfloor heating switch/l7/set' with data 'OFF'
Debug 2023-11-24 12:22:31Publishing 'set' 'state' to 'Underfloor heating switch'
Debug 2023-11-24 12:22:31Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0],"type":"Buffer"},"datatype":1,"dp":6}],"seq":20736}' from endpoint 1 with groupID 0
Info 2023-11-24 12:22:31MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":72,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"OFF","state_l7":"ON","state_l8":"OFF"}'
Info 2023-11-24 12:22:31MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":72,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"OFF","state_l7":"OFF","state_l8":"OFF"}'
Debug 2023-11-24 12:22:31Received Zigbee message from 'Underfloor heating switch', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0],"type":"Buffer"},"datatype":1,"dp":7}],"seq":20992}' from endpoint 1 with groupID 0
Info 2023-11-24 12:22:31MQTT publish: topic 'zigbee2mqtt/Underfloor heating switch', payload '{"linkquality":76,"state_l1":"OFF","state_l2":"OFF","state_l3":"OFF","state_l4":"OFF","state_l5":"OFF","state_l6":"OFF","state_l7":"OFF","state_l8":"OFF"}'`

**Comment by Koenkk** [comment]
I think we can solve this by creating a new `valueConverter` (instead of `valueConverter.onOff`) which has a sleep inside.

**Comment by bszczepanik** [comment]
Wouldn't this cause all requests to be delayed by the same amount of time? All requests will be sent at the same time, but delayed.

**Comment by Koenkk** [comment]
I don't think so, they should be send sequentially

File: src/lib/tuya.ts [hunk 1]
```diff
@@ -440,6 +440,13 @@ export const valueConverter = {
     trueFalseEnum0: valueConverterBasic.trueFalse(new Enum(0)),
     trueFalseEnum1: valueConverterBasic.trueFalse(new Enum(1)),
     onOff: valueConverterBasic.lookup({'ON': true, 'OFF': false}),
+    onOffWithDelay: {
+        to: async (value: string) => {
+            await utils.sleep(200);
```
Comment (bszczepanik) [inline src/lib/tuya.ts:445]: I put delay here, but that caused that all requests are delayed. It should look like `utils.sleep(transitiont * 200)` but I don't know how proper declare `transition`

Comment (Koenkk) [reply src/lib/tuya.ts:445]: What do you mean with `transitiont`?

Comment (bszczepanik) [reply src/lib/tuya.ts:445]: When I will change switch in Z2M for more then one entity at one time, I want to set some number like 0, 1, 2, 3 for each. I will get delay time such 0 * 200 ms, 1 * 200 ms, 2 * 200 ms and so on.

Comment (Koenkk) [reply src/lib/tuya.ts:445]: If you change `to: async (value: string) => {` to `to: async (value: string, meta) => {`, you can get the message with `meta.message`, from that you can count the number of `state_` and create the timeout based on that.

**Comment by bszczepanik** [comment]
That code works for me. I am not sure is it correct to put value clear function to on event function. It works but I don't know is it good way.

**Comment by Koenkk** [comment]
Instead of using the globalStore, did you experiment with using the `meta.message` instead?

**Comment by bszczepanik** [comment]
I checked this device once again with default onoff functions. I realived that after update Z2M and Home Assistan there is no problem with freezing channels. That configuration works correct.

**Pull request merged**